erpnext/public/js/utils.js

```diff
@@ -293,7 +293,7 @@ $.extend(erpnext.utils, {
 	},
 	copy_parent_value_in_all_row: function(doc, dt, dn, table_fieldname, fieldname, parent_fieldname) {
 		var d = locals[dt][dn];
-		if(d[fieldname]){
+		if(d[parent_fieldname]){
```

Comment:
- `parent_fieldname` must pertain to a field that is not in the table, and is in the outer form. 
- `d` here contains the child row, which will only have `fieldname`.

Your fix is checking for the parent field in the child row, which is incorrect. Am I missing something ?

Reply:
Hi @marination , Thanks for your reply.

I'm not very sure what's the original developer's purpose by writing the `copy_parent_value_in_all_row`. But if we take a look of its sibling function `copy_value_in_all_rows` on line 153 same file:

![image](https://user-images.githubusercontent.com/30763348/114380098-c0d20880-9bdd-11eb-97c8-097eed446ec2.png)

It assumes that `field_name` exists both on outer form and child table. And it only pastes the value in if `d[field_name]` has value.

What I want to achive by using `copy_parent_value_in_all_row` is that it can take outer form's field value and overrides to every rows in child table.

My use case in custom script:
![image](https://user-images.githubusercontent.com/30763348/114381034-ca0fa500-9bde-11eb-8052-aab9454196c5.png)

In my case, since all purchase orders submitted with schedule_date in outer form will use `copy_value__in_all_rows` to add schedule_date in items table. We don't use schedule date field that much, but users would like to set `expected_delivery_date` on items table, which is a field not existing in outer form.

Hence I have modified this change to achieve my goal which is to prompt user to update an ETA which overrides `schedule_date` in outer form and all child table rows plus set the value in "expected_delivery_date" in all child table rows as well.

I would suggest to rename this function as **overrides_parent_value_in_all_rows** and then remove the if statement validation, just loop through each item and overrides the `parent_fieldname` to `d[field_name]`.

Sorry I realized my reply might be too long and probably doesn't make sense.
Please leave any your questions or feedbacks. Thank you! 😊

Reply:
@AndyOverLord Makes sense. Just some poorly defined utilities. Will try to clean up the rest later. Your changes look fine.